### PR TITLE
Large commit, lot of prettification, some small changes

### PR DESCRIPTION
**Pull request by sumitsahrawat**
Sorry for such a large commit. :smiling_imp:.

Major changes are

- Hyperlinking
- Line wrapping
- Less bullets, more code-blocks
- Added tony_parametricity.pdf (why not share directly)

### CODE REVIEW

**Comment by bitemyapp** [comment]
This is a *lot*, thank you!

Few things:

Looks like there's a merge conflict?

Also, why did you add line-breaks?

**Comment by sumitsahrawat** [comment]
I'll resolve the merge conflicts. I don't have much experience with pull requests, so I didn't take a look at this page to check for merge conflicts.

I added line breaks because editing a file with long lines is difficult in some editors.
For example I use vim, but only for small changes from the terminal, and thus don't know much about customizing it. It wraps lines properly, but going down takes you to the next line. Using emacs, wrapping all paragraphs to 72 characters was easy, so I did it. :smile:

Also, there are some small changes that you might have missed. I'll see if I can make a changelog while resolving conflicts.

And, you're welcome. The guide is a real :gem:.

File: README.md [hunk 1]
```diff
@@ -113,22 +150,28 @@ run `cabal` in a terminal, your shell will run the up-to-date version in your
...
+```
 
 If you don't know what your shell is, more than likely, your shell is Bash. If
 you use Bash, the file you will edit is `~/.bashrc`. If you use Z-shell, the
 file is `~/.zshrc`. You can run the following command to find out what your
 shell is.
 
-    < /etc/passwd grep $(whoami) | awk -F/ '{print $NF}'
+```bash
+echo $SHELL | xargs basename
```
Comment (sumitsahrawat) [inline README.md]: I'm don't consider myself a shell expert, so you might want to see whether both the commands work the same.

File: README.md [hunk 2]
```diff
@@ -158,144 +202,187 @@ Download the latest binary distributions for cabal and ghc:
...
 ## FAQ and working with Cabal
 
 ### Fantastic FAQ
 
-In addition to being an amazing guide for all kinds of things such as GADTs, this also covers some useful basics for Cabal
+In addition to being an amazing guide for all kinds of things such as GADTs,
+this also covers some useful basics for Cabal
 
-- http://dev.stephendiehl.com/hask/
+- [What I wish I knew when learning Haskell](http://dev.stephendiehl.com/hask/)
```
Comment (sumitsahrawat) [inline README.md]: Added github link.

File: README.md [hunk 3]
```diff
@@ -303,86 +390,108 @@ The other thing to keep in mind is that due to the way the $pkg parameter gets i
...
+- [More points for lazy evaluation](http://augustss.blogspot.hu/2011/05/more-points-for-lazy-evaluation-in.html)
 
-- http://alpmestan.com/posts/2013-10-02-oh-my-laziness.html
+- [Oh my laziness!](http://alpmestan.com/posts/2013-10-02-oh-my-laziness.html)
 
-- http://stackoverflow.com/questions/13042353/does-haskell-have-tail-recursive-optimization
+- SO question '[Does haskell have laziness?](http://stackoverflow.com/questions/13042353/does-haskell-have-tail-recursive-optimization)'
 
-- http://www.slideshare.net/tibbe/reasoning-about-laziness
+- [Johan Tibell](https://github.com/tibbe)'s slides from a talk on
```
Comment (sumitsahrawat) [inline README.md]: A guy with a good style guide :+1:.

File: README.md [hunk 4]
```diff
@@ -397,226 +506,251 @@ let (v : _) = a
...
 
-- http://www.haskell.org/haskellwiki/Category_theory good links to other resources
+- [Category Theory](http://www.haskell.org/haskellwiki/Category_theory) on
+  haskellwiki, also has good links to other resources
 
-- http://science.raphael.poss.name/categories-from-scratch.html includes practical examples
+- [Categories from scratch](http://science.raphael.poss.name/categories-from-scratch.html), Includes some practical examples.
 
-- https://www.google.com/search?q=Awodey+Category+Theory the standard text along with MacLane
+- [Awodey](http://ukcatalogue.oup.com/product/9780199237180.do) and
```
Comment (sumitsahrawat) [inline README.md]: Links to Awodey and MacLane.

**Comment by sumitsahrawat** [comment]
It seems to be fixed now :tada:.

**Comment by bitemyapp** [comment]
@sumitsahrawat thank you so much, this is huge!

:panda_face: :bear: :honey_pot: :beers: 

If you're ever near Austin, TX, let me know and I'll get you dinner :)

**Comment by sumitsahrawat** [comment]
Thanks for the offer.
I'm from India, but if the chance ever comes, I'll remember it.

**Pull request merged**